src/RA_Dlg_MemBookmark.cpp

```diff
@@ -654,14 +654,17 @@ void Dlg_MemBookmark::WriteFrozenValue(const MemBookmark & Bookmark)
     char buffer[32];
     sprintf_s(buffer, sizeof(buffer), "%0*x", width, Bookmark.Value());
 
-    for (unsigned int i = 0; i < strlen(buffer); i++)
+    if(std::strlen(buffer) > 0)
```

Comment:
why would buffer be empty? if this is necessary, suggest "if (*buffer)", we don't need to iterate the entire string.

Reply:
Sometimes it came out at nullptr when testing file i/o for local achievements.

Reply:
buffer can't be nullptr as it's an array on the stack. if it contains nullptr, that means the sprintf_s failed. suggest checking the return value of sprintf_s vs running strlen.

Reply:
it may not even contain nullptr in a release build (uninitialized memory in buffer). definitely recommend checking the return value.